### PR TITLE
Use AVM for log analytics and application insights.

### PR DESCRIPTION
**Pull request by tonybaloney**
Provisions the same resources but using the AVM resources. Replaces the 2 custom bicep files that existed for provisioning the resources with the custom private network flag.

### CODE REVIEW

File: infra/core/monitor/monitoring.bicep [hunk 1]
```diff
@@ -7,33 +7,43 @@ param tags object = {}
...
-module logAnalytics 'loganalytics.bicep' = {
+module logAnalytics 'br/public:avm/res/operational-insights/workspace:0.4.0' = {
   name: 'loganalytics'
   params: {
     name: logAnalyticsName
     location: location
     tags: tags
+    skuName: 'PerGB2018'
+    dataRetention: 30
     publicNetworkAccessForIngestion: publicNetworkAccess
```
Comment (pamelafox) [inline infra/core/monitor/monitoring.bicep:18]: I'm surprised that the avm has a parameter named publicNetworkAccessForIngestion ? How is that working?

Comment (tonybaloney) [reply infra/core/monitor/monitoring.bicep:18]: It's a parameter of the log analytics workspace and the application insights account anyway https://github.com/Azure/bicep-registry-modules/blob/main/avm/res/operational-insights/workspace/main.bicep#L192-L193

Comment (pamelafox) [reply infra/core/monitor/monitoring.bicep:18]: Ohh, I thought it was one we made up ("ingestion" threw me off). Got it, thanks

**Comment by pamelafox** [comment]
Which envs have you tested with, both public and vnet? I want to figure out what still needs testing.

**Comment by tonybaloney** [comment]
After doing a diff on the deployment, the only thing this changes is `forceCmkForQuery` to true (previously unspecified) for the log analytics workspace which improves security and should have no impact on most users.

**Review by pamelafox — APPROVED** [review]

**Comment by pamelafox** [comment]
Okay, thanks for diff'ing! @mattgotteiner This will be the first move to AVM but the hope is to move all modules to AVM, to reduce Bicep modules in our code and inherit best practices by default from AVM.

**Pull request merged**